.gitignore

```diff
@@ -0,0 +1,218 @@
+
+# Created by https://www.toptal.com/developers/gitignore/api/linux,macos,dotenv,windows,intellij,visualstudiocode
+# Edit at https://www.toptal.com/developers/gitignore?templates=linux,macos,dotenv,windows,intellij,visualstudiocode
+
+### dotenv ###
+.env
+
+### Intellij ###
+# Covers JetBrains IDEs: IntelliJ, RubyMine, PhpStorm, AppCode, PyCharm, CLion, Android Studio, WebStorm and Rider
+# Reference: https://intellij-support.jetbrains.com/hc/en-us/articles/206544839
+
```

Comment:
This `.gitignore` file is huge and includes all kinds of things that one wouldn't expect to be in a checkout of this repo.

IMO developers should have a global `.gitignore_global` that ignores all kinds of things specific to their setup and not put everything inside a project's `.gitignore`.

Could you please only ignore the files that would credibly be inside a checkout of this repo and which we don't want to track?